Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.AwsSecurityCredential.cs

```diff
@@ -0,0 +1,144 @@
+﻿/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Json;
+using Google.Apis.Util;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    public partial class AwsExternalAccountCredential
+    {
+        /// <summary>
+        /// Represents AWS security credentials which are used to sign
+        /// the subject token.
+        /// </summary>
+        private sealed class AwsSecurityCredentials
+        {
+            private const string AccessKeyIdEnvVar = "ACCESS_ACCESS_KEY_ID";
+            private const string SecretAccessKeyEnvVar = "AWS_SECRET_ACCESS_KEY";
+            private const string TokenEnvVar = "AWS_SESSION_TOKEN";
+
+            internal string AccessKeyId { get; }
+
+            internal string SecretAccessKey { get; }
+
+            internal string Token { get; }
+
+            private AwsSecurityCredentials(string accessKeyId, string secretAccessKey, string token)
+            {
+                AccessKeyId = accessKeyId;
+                SecretAccessKey = secretAccessKey;
+                Token = token;
+            }
+
+            /// <summary>
+            /// Fetches the AWS security credentials as per https://google.aip.dev/auth/4117#determining-the-subject-token-in-aws.
+            /// </summary>
+            internal static async Task<AwsSecurityCredentials> FetchAsync(AwsMetadataServerClient metadataClient, string credentialUrl) =>
+                MaybeFromEnvVars()
+                ?? await MaybeFromMetadataAsync(metadataClient, credentialUrl).ConfigureAwait(false)
+                ?? throw new InvalidOperationException($"Couldn't determine the AWS security credentials." +
+                    $"At least one of the following environment variables was not set: {AccessKeyIdEnvVar}, {SecretAccessKeyEnvVar}." +
+                    $"And the {nameof(SecurityCredentialsUrl)} was not specified in the credential configuration" +
+                    "or no security credentials couldn't be obtained from it.");
```

Comment:
couldn't => could (or remove the "no" from earlier in the line - basically let's just have a single negative)